Snowflake.Data.Tests/IntegrationTests/SFPutGetTest.cs

```diff
@@ -474,29 +517,41 @@ private static bool IsCompressedByTheDriver()
 
                 // Upload file
                 command.CommandText = putQuery;
-                var reader = command.ExecuteReader();
-                Assert.IsTrue(reader.Read());
-
-                // Check file status
-                Assert.AreEqual(expectedStatus.ToString(),
-                    reader.GetString((int)SFResultSet.PutGetResponseRowTypeInfo.ResultStatus));
-                // Check source and destination compression type
-                if (t_autoCompress)
+                try
                 {
-                    Assert.AreEqual(t_sourceCompressionType,
-                        reader.GetString((int)SFResultSet.PutGetResponseRowTypeInfo.SourceCompressionType));
-                    Assert.AreEqual(t_destCompressionType,
-                        reader.GetString((int)SFResultSet.PutGetResponseRowTypeInfo.DestinationCompressionType));
+                    var reader = command.ExecuteReader();
+                    Assert.IsTrue(reader.Read());
+                    // Checking query id when reader succeeded
+                    queryId = ((SnowflakeDbDataReader)reader).GetQueryId();
+                    // Checking if query Id is provided on the command level as well
+                    Assert.AreEqual(queryId, ((SnowflakeDbCommand)command).GetQueryId());
+                    // Check file status
+                    Assert.AreEqual(expectedStatus.ToString(),
+                        reader.GetString((int)SFResultSet.PutGetResponseRowTypeInfo.ResultStatus));
+                    // Check source and destination compression type
+                    if (t_autoCompress)
+                    {
+                        Assert.AreEqual(t_sourceCompressionType,
+                            reader.GetString((int)SFResultSet.PutGetResponseRowTypeInfo.SourceCompressionType));
+                        Assert.AreEqual(t_destCompressionType,
+                            reader.GetString((int)SFResultSet.PutGetResponseRowTypeInfo.DestinationCompressionType));
+                    }
+                    else
+                    {
+                        Assert.AreEqual(SFFileCompressionTypes.NONE.Name,
+                            reader.GetString((int)SFResultSet.PutGetResponseRowTypeInfo.SourceCompressionType));
+                        Assert.AreEqual(SFFileCompressionTypes.NONE.Name,
+                            reader.GetString((int)SFResultSet.PutGetResponseRowTypeInfo.DestinationCompressionType));
+                    }
+                    Assert.IsNull(reader.GetString((int)SFResultSet.PutGetResponseRowTypeInfo.ErrorDetails));
                 }
-                else
+                catch (SnowflakeDbException e)
```

Comment:
We need this catch to verify if in case of any failure the command regardless of the outcome contains query id but I will move this catch up and not consume but rethrow

Reply:
Also when PUT failed it either didn't throw or if it did some or all asserts wouldn't take place but I'll reorganize it so it is less prone to hiding issues.